client/server/EngineModel.js

```diff
@@ -0,0 +1,72 @@
+"use strict";
+Object.defineProperty(exports, "__esModule", { value: true });
+const vscode_languageserver_1 = require("vscode-languageserver");
+var fs = require('fs');
+var xml2js = require('xml2js');
+var parser = new xml2js.Parser();
+class EngineModel {
+    constructor(path, completionItems) {
+        this.path = path;
+        this.completionItems = completionItems;
+        this.loadPredefinedObjects();
+    }
+    loadPredefinedObjects() {
+        // load predefined objects
+        if (fs.existsSync(this.path)) {
+            this.parsePredefinedObjects(this.path, this.completionItems);
+        }
+        else {
+            console.warn(`File “${this.path}” doesn't exist.`);
```

Comment:
Shouldn't this be an error rather than a warning? I guess lots of things will go wrong afterwards.

Reply:
Nothing will go wrong. We just ignore the parsing of the file. However, maybe we should warn user directly with a notification? The path is added directly by the user, after all.